polling_stations/apps/addressbase/helpers.py

```diff
@@ -0,0 +1,90 @@
+from pollingstations.models import (PollingDistrict, ResidentialAddress,
+                                    PollingStation)
+from addressbase.models import Address
+
+
+def centre_from_points_qs(qs):
+    if not qs:
+        return None
+
+    if len(qs) == 1:
+        return qs[0].location
+
+    base_point = qs[0].location
+    poly = base_point.union(qs[1].location)
+    for m in qs:
+        poly = poly.union(m.location)
+
+    return poly.centroid
+
+
+def district_contains_all_points(district, points):
+    return all([district.area.contains(p) for p in points])
+
+
+def postcodes_not_contained_by_district(district):
+    data = {
+        'not_contained': [],
+        'total': 0
+    }
+
+    for postcode in Address.objects.postcodes_for_district(district):
+        points = Address.objects.points_for_postcode(postcode)
+        data['total'] += 1
+        if not district_contains_all_points(district, points):
+            data['not_contained'].append(postcode)
+    return data
+
+
+def make_addresses_for_postcode(postcode, council_id):
+    addresses = Address.objects.filter(postcode=postcode)
+    created = 0
+    for address in addresses:
+        try:
+            district = PollingDistrict.objects.get(
+                area__covers=address.location, council_id=council_id)
+        except PollingDistrict.DoesNotExist:
+            # Chances are this is on the edge of the council area, and
+            # we don't have data for the are the property is in
+            continue
+        except PollingDistrict.MultipleObjectsReturned:
+            # This is normally causes by districts the overlap
+            # Because we have no way of knowing what the correct district is,
+            # we need to ignore this address
```

Comment:
OK, so what this is going to do is just exclude the one problematic address (i.e: if there are n addresses covered by this postcode, and one sits on an overlap, we insert n-1 addresses). This isn't **wrong** as such, but if you are the person who lives at that address, its a bit confusing as there is no clear "what do I do next" action for them at this point. One really hacky "fix" would be to insert that address with an intentionally invalid station id reference (which I think should send them to "we don't know - call your council"). Alternatively it could be dealt with in a more generic fashion with the "my address is not in the list" button proposed in https://github.com/DemocracyClub/UK-Polling-Stations/pull/326#issuecomment-226967544 (which also solves issues like "my address is newbuild and was registered with Royal Mail/Highways Agency since the last AddressBase was published, etc)
Probably not a blocker for this PR though.